minicbor-derive/src/fields.rs

```diff
@@ -63,7 +63,7 @@ impl Fields {
             }
         }
 
-        fields.sort_unstable_by_key(|f| f.index.val());
+        fields.sort_unstable_by_key(|f| f.index.val_for_sorting());
```

Comment:
Why is this required?

Reply:
The comment on the function was incomplete; now fixed.

My understanding of what this sorting does here is to ensure that when serialization happens field by field, the resulting map is in preferred encoding; `val_for_sorting()` creates an order that is identical with the order of the later encoded items (which was trivial for u32, but is not when negative items are allowed).